### PR TITLE
Add Taocpp taopq 20200222

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **taocpp-taopq/20200222**

/cc @d-frey

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/taocpp-taopq/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        bin_path = os.path.join("bin", "test_package")
```
Comment (madebr) [inline recipes/taocpp-taopq/all/test_package/conanfile.py]: Indent with `tools.cross_building`

Comment (uilianries) [reply recipes/taocpp-taopq/all/test_package/conanfile.py]: ctrl-c ctrl-v curse. Thanks for reviewing!

Comment (madebr) [reply recipes/taocpp-taopq/all/test_package/conanfile.py]: No problem. Looking at code is easier then writing :smile:

File: recipes/taocpp-taopq/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        extracted_dir = glob.glob("taopq-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["TAOPQ_BUILD_TESTS"] = False
+            self._cmake.definitions["TAOPQ_INSTALL_DOC_DIR"] = "licenses"
+            if self.settings.os == 'Windows' and self.settings.compiler == 'Visual Studio':
+                self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
```
Comment (danimtb) [inline recipes/taocpp-taopq/all/conanfile.py]: This is only available since cmake 3.4

Comment (uilianries) [reply recipes/taocpp-taopq/all/conanfile.py]: Done! I've update the CMakeLists.txt wrapper.

**Comment by danimtb** [comment]
Seems there was a timeout contacting Artifactory due to some network latency. Relaunching the CI

**Pull request merged**